src/main/java/teammates/ui/webapi/CreateFeedbackResponseCommentAction.java

```diff
@@ -158,38 +281,77 @@ public JsonResult execute() throws InvalidHttpRequestBodyException, InvalidOpera
             throw new InvalidHttpParameterException("Unknown intent " + intent);
         }
 
-        String feedbackQuestionId = response.getFeedbackQuestionId();
-        String feedbackSessionName = response.getFeedbackSessionName();
-
-        FeedbackResponseCommentAttributes feedbackResponseComment = FeedbackResponseCommentAttributes
-                .builder()
-                .withCourseId(courseId)
-                .withFeedbackSessionName(feedbackSessionName)
-                .withCommentGiver(email)
-                .withCommentText(commentText)
-                .withFeedbackQuestionId(feedbackQuestionId)
-                .withFeedbackResponseId(feedbackResponseId)
-                .withGiverSection(response.getGiverSection())
-                .withReceiverSection(response.getRecipientSection())
-                .withCommentFromFeedbackParticipant(isFromParticipant)
-                .withCommentGiverType(commentGiverType)
-                .withVisibilityFollowingFeedbackQuestion(isFollowingQuestionVisibility)
-                .withShowCommentTo(comment.getShowCommentTo())
-                .withShowGiverNameTo(comment.getShowGiverNameTo())
-                .build();
-
-        FeedbackResponseCommentAttributes createdComment;
+        FeedbackResponseComment feedbackResponseComment = new FeedbackResponseComment(feedbackResponse, email,
+                commentGiverType, feedbackResponse.getGiverSection(), feedbackResponse.getRecipientSection(), commentText,
+                isFollowingQuestionVisibility, isFromParticipant, comment.getShowCommentTo(), comment.getShowGiverNameTo(),
+                email);
         try {
-            createdComment = logic.createFeedbackResponseComment(feedbackResponseComment);
-        } catch (EntityDoesNotExistException e) {
-            throw new EntityNotFoundException(e);
-        } catch (EntityAlreadyExistsException e) {
-            throw new InvalidOperationException(e);
+            FeedbackResponseComment createdComment = sqlLogic.createFeedbackResponseComment(feedbackResponseComment);
+            HibernateUtil.flushSession();
+            return new JsonResult(new FeedbackResponseCommentData(createdComment));
         } catch (InvalidParametersException e) {
             throw new InvalidHttpRequestBodyException(e);
+        } catch (EntityAlreadyExistsException e) {
+            return new JsonResult(e.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
```

Comment:
I believe we should throw `InvalidOperationException` here?

Reply:
ah you're right, changed!